crates/compiler/mono/src/drop_specialization.rs

```diff
@@ -87,16 +88,46 @@ fn specialize_drops_stmt<'a, 'i>(
             }
 
             match expr {
-                Expr::Call(_) | Expr::Struct(_) => {
-                    // TODO perhaps allow for some e.g. lowlevel functions to be called if they cannot modify the RC of the symbol.
+                Expr::Call(Call {
+                    call_type,
+                    arguments,
+                }) => {
+                    match call_type {
+                        CallType::ByName { name, .. }
+                            if (LowLevelWrapperType::from_symbol(name.name())
+                                == LowLevelWrapperType::CanBeReplacedBy(
+                                    LowLevel::ListGetUnsafe,
+                                )) =>
+                        {
+                            environment.add_list_child(arguments[0], *binding, &arguments[1]);
+
+                            alloc_let_with_continuation!(environment)
+                        }
+                        CallType::LowLevel {
+                            op: LowLevel::ListGetUnsafe,
+                            ..
+                        } => {
+                            environment.add_list_child(arguments[0], *binding, &arguments[1]);
+
+                            alloc_let_with_continuation!(environment)
+                        }
+                        _ => {
+                            // TODO perhaps allow for some e.g. lowlevel functions to be called if they cannot modify the RC of the symbol.
 
-                    // Calls can modify the RC of the symbol.
-                    // If we move a increment of children after the function,
-                    // the function might deallocate the child before we can use it after the function.
-                    // If we move the decrement of the parent to before the function,
-                    // the parent might be deallocated before the function can use it.
-                    // Thus forget everything about any increments.
+                            // Calls can modify the RC of the symbol.
+                            // If we move a increment of children after the function,
+                            // the function might deallocate the child before we can use it after the function.
+                            // If we move the decrement of the parent to before the function,
+                            // the parent might be deallocated before the function can use it.
+                            // Thus forget everything about any increments.
```

Comment:
let's make a separate function that given a lowlevel, may or may not add additional RC operations. I think that'll be a bit cleaner.

Reply:
What do you mean exactly?
I purposefully put it here as the behaviour is exceptional

Reply:
I understand, I thought maybe it would be nice to have a helper like `try_lowlevel(expr: &Expr) -> Option<LowLevel>` and then match on that. the whole matching on named and lowlevel calls is very noisy, and duplicates logic.

Reply:
I created `replace_lowlevel_wrapper` to fix this problem, which solves `is_replaced_by` as well.